ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -101,6 +101,17 @@ export function WorkflowLogsViewer({workflow, nodeId, initialPodName, container,
     useEffect(() => {
         setError(null);
         setLoaded(false);
+
+        if (archived) {
+            // replace nodeId with a nodeId which is associated with the podName, so we can also switch bewteen archived logs if a differnet node was selected.
```

Comment:
```suggestion
            // use a nodeId of a pod, so we can switch between archived logs if a different node was selected
```
- fix multiple typos
- shorten and clarify